sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumerAsyncClient.java

```diff
@@ -163,52 +136,122 @@ public String getConsumerGroup() {
+     * @param startingPosition Position within the Event Hub partition to begin consuming events.
+     *
+     * @return A stream of events for this partition. If a stream for the events was opened before, the same position
+     *     within that partition is returned. Otherwise, events are read starting from {@code startingPosition}.
+     *
+     * @throws NullPointerException if {@code partitionId}, {@code startingPosition}, {@code receiveOptions} is null.
+     * @throws IllegalArgumentException if {@code partitionId} is an empty string.
+     */
+    public Flux<PartitionEvent> receiveFromPartition(String partitionId, EventPosition startingPosition) {
+        return receiveFromPartition(partitionId, startingPosition, defaultReceiveOptions);
+    }
+
     /**
-     * Begin consuming events from a single partition starting at {@link #getStartingPosition()} until there are no more
-     * subscribers.
+     * Begin consuming events from a single partition starting at {@code startingPosition} with a set of
+     * {@link ReceiveOptions receive options}.
+     *
+     * <ul>
+     * <li>If receive is invoked where {@link ReceiveOptions#getOwnerLevel()} has a value, then Event Hubs service will
+     * guarantee only one active consumer exists per partitionId and consumer group combination. This consumer is
+     * sometimes referred to as an "Epoch Consumer."</li>
+     * <li>Multiple consumers per partitionId and consumer group combination can be created by not setting
+     * {@link ReceiveOptions#getOwnerLevel()} when creating consumers. This non-exclusive consumer is sometimes
+     * referred to as a "Non-Epoch Consumer."</li>
+     * </ul>
      *
      * @param partitionId Identifier of the partition to read events from.
+     * @param startingPosition Position within the Event Hub partition to begin consuming events.
+     * @param receiveOptions Options when receiving events from the partition.
+     *
      * @return A stream of events for this partition. If a stream for the events was opened before, the same position
-     *     within that partition is returned. Otherwise, events are read starting from {@link #getStartingPosition()}.
+     *     within that partition is returned. Otherwise, events are read starting from {@code startingPosition}.
      *
-     * @throws NullPointerException if {@code partitionId} is null.
+     * @throws NullPointerException if {@code partitionId}, {@code startingPosition}, {@code receiveOptions} is null.
      * @throws IllegalArgumentException if {@code partitionId} is an empty string.
      */
-    public Flux<PartitionEvent> receive(String partitionId) {
-        if (partitionId == null) {
+    public Flux<PartitionEvent> receiveFromPartition(String partitionId, EventPosition startingPosition,
+        ReceiveOptions receiveOptions) {
+        if (Objects.isNull(partitionId)) {
             return Flux.error(logger.logExceptionAsError(new NullPointerException("'partitionId' cannot be null.")));
```

Comment:
azure-core has a convenience method `fluxError` for this.

Reply:
Fixed